app/src/main/java/com/rightpoint/emojipuzzler/puzzle/PuzzleViewModel.kt

```diff
@@ -0,0 +1,51 @@
+package com.rightpoint.emojipuzzler.puzzle
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.rightpoint.domain.IDomainClient
+import com.rightpoint.domain.models.EmojiPuzzle
+import com.rightpoint.emojipuzzler.EmojiPuzzleApplication
+import kotlinx.coroutines.*
+
+class PuzzleViewModel: ViewModel() {
+
+    private var domain: IDomainClient = EmojiPuzzleApplication.domainClient
+
+    var puzzle: EmojiPuzzle? = null
+
+    var feedBackLiveData: MutableLiveData<String> = MutableLiveData()
+    var currentEmojiPuzzle: MutableLiveData<String> = MutableLiveData()
+    var loading: MutableLiveData<Boolean> = MutableLiveData()
+
```

Comment:
#nth - "feedBackLiveData", "currentEmojiPuzzle", "loading" should all be immutable: val.